recipes/openfoam/build.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+export FOAM_DIR_NAME="${SRC_DIR}" #/openfoam-OpenFOAM-${PKG_VERSION}"
+
+ln -s $CPP $BUILD_PREFIX/bin/cpp
+
+sed -i 's/gcc\$(COMPILER_VERSION)/\$(GCC)\$(COMPILER_VERSION)/g' ${FOAM_DIR_NAME}/wmake/rules/General/Gcc/c
+sed -i 's/g++\$(COMPILER_VERSION)/\$(GXX)\$(COMPILER_VERSION)/g' ${FOAM_DIR_NAME}/wmake/rules/General/Gcc/c++
+sed -i 's/= ar/?= ar/g' ${FOAM_DIR_NAME}/wmake/rules/General/general
+sed -i 's/= ranlib/?= ranlib/g' ${FOAM_DIR_NAME}/wmake/rules/General/general
+sed -i 's/= cpp/?= cpp/g' ${FOAM_DIR_NAME}/wmake/rules/General/general
+sed -i 's/= ld/?= ld/g' ${FOAM_DIR_NAME}/wmake/rules/General/general
+
+# source foam dot file throws error if not compiled
+# modify the the output folder of the binaries 
+sed -i 's/\$WM_PROJECT_DIR\/platforms\/\$WM_OPTIONS/\$\{PREFIX\}/g' ${FOAM_DIR_NAME}/etc/config.sh/settings
+source "${FOAM_DIR_NAME}/etc/bashrc" WM_COMPILER=Gcc WM_MPLIB=MPICH || true
+export CONFIGSHDIR=${FOAM_DIR_NAME}/etc/config.sh
+
+# change scotch version to the conda version
+sed -i 's/^SCOTCH_VERSION=.*/SCOTCH_VERSION=scotch-system/g' ${CONFIGSHDIR}/scotch
+sed -i 's/^export SCOTCH_ARCH_PATH=.*/export SCOTCH_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/scotch
+
+sed -i 's/^ZOLTAN_VERSION=.*/ZOLTAN_VERSION=zoltan-system/g' ${CONFIGSHDIR}/zoltan
+sed -i 's/^export ZOLTAN_ARCH_PATH=.*/export ZOLTAN_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/zoltan
+
+# change kahip version to the conda version
+sed -i 's/^KAHIP_VERSION=.*/KAHIP_VERSION=kahip-system/g' ${CONFIGSHDIR}/kahip
+sed -i 's/^export KAHIP_ARCH_PATH=.*/export KAHIP_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/kahip
+
+# change metis version to the conda version
+sed -i 's/^METIS_VERSION=.*/METIS_VERSION=metis-system/g' ${CONFIGSHDIR}/metis
+sed -i 's/^export METIS_ARCH_PATH=.*/export METIS_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/metis
+
+# change petsc version to the conda version
+sed -i 's/^petsc_version=.*/petsc_version=petsc-system/g' ${CONFIGSHDIR}/petsc
+sed -i 's/^export PETSC_ARCH_PATH=.*/export PETSC_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/petsc
+
+# change hypre version to the conda version
+sed -i 's/^hypre_version=.*/hypre_version=hypre-system/g' ${CONFIGSHDIR}/hypre
+sed -i 's/^export HYPRE_ARCH_PATH=.*/export HYPRE_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/hypre
+
+# change boost and cgal version to the conda version
+sed -i 's/^boost_version=.*/boost_version=boost-system/g' ${CONFIGSHDIR}/CGAL
+sed -i 's/^export BOOST_ARCH_PATH=.*/export BOOST_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/CGAL
+sed -i 's/^cgal_version=.*/cgal_version=cgal-system/g' ${CONFIGSHDIR}/CGAL
+sed -i 's/^export CGAL_ARCH_PATH=.*/export CGAL_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/CGAL
+
+# change fftw version to the conda version
+sed -i 's/^fftw_version=.*/fftw_version=fftw-system/g' ${CONFIGSHDIR}/FFTW
+sed -i 's/^export FFTW_ARCH_PATH=.*/export FFTW_ARCH_PATH=${PREFIX}/g' ${CONFIGSHDIR}/FFTW
+
+#
+echo "cFLAGS += -I ${BUILD_PREFIX}/include" >> "${FOAM_DIR_NAME}/wmake/rules/linux64Gcc/c"
```

Comment:
Here you shouldn't be using BUILD_PREFIX but simply PREFIX. It may even be better if you used the CXXFLAGS as set in the environment.

Reply:
Changed that, thanks for pointing oui, now using CFLAGS, CXXFLAGS and LDFLAGS instead